api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -516,6 +523,43 @@ message HttpConnectionManager {
   bool strip_matching_host_port = 39;
 }
 
+// The configuration to customize local reply returned by Envoy.
+message LocalReplyConfig {
+  // Configuration of list of mappers which allows to filter and change local response.
+  // The mappers will be checked by the specified order until one is matched.
+  repeated ResponseMapper mappers = 1;
+
+  // The configuration to form response body from the :ref:`command operators <config_access_log_command_operators>`
+  // and to specify response content type as one of: plain/text or application/json.
+  // If the matched `mapper <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.LocalReplyConfig.mapper>`
+  // has a valid `format <envoy_v3_api_field_extensions.filters.network.http_connection_manager.v3.ResponseMapper.format>`,
+  // it will be used, this one will not be used.
+  config.core.v3.SubstitutionFormatString format = 2;
+}
+
+// The configuration to filter and change local response.
+message ResponseMapper {
+  // Filter to determine if the response should be changed.
+  config.accesslog.v3.AccessLogFilter filter = 1 [(validate.rules).message = {required: true}];
+
+  // Rewriter defines how the response should be changed.
+  ResponseRewriter rewriter = 2 [(validate.rules).message = {required: true}];
+
+  // The configuration to form new response body from the :ref:`command operators <config_access_log_command_operators>`
+  // and to specify response content type as one of: plain/text or application/json.
+  // It is only used when this mapper is matched.
+  config.core.v3.SubstitutionFormatString format = 3;
+}
+
+// Configuration on how to change the response.
+message ResponseRewriter {
```

Comment:
Do we need this message? Can we just inline these fields in the mapper? It's not clear to me that we need this to be a separate message. Thoughts?